### PR TITLE
Fixes #1860. Fix roll failures

### LINKED ISSUE #1860: Failures on [co19] Roll co19 to e07466f68ea53c18c85024e364b936d00b4cd06e
There are new test failures on [[co19] Roll co19 to e07466f68ea53c18c85024e364b936d00b4cd06e](https://dart-review.googlesource.com/c/sdk/+/283800).

The tests
```
co19/LanguageFeatures/Patterns/shared_case_scope_A01_t04 MissingCompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/type_inference_A11_t02 CompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/type_inference_A11_t03 CompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/type_inference_A12_t01 CompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/type_inference_A13_t02 CompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/type_inference_A13_t03 CompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/type_inference_A14_t02 CompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/type_inference_A14_t03 CompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/type_inference_A16_t02 CompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/type_inference_A16_t03 CompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/type_inference_A17_t05 CompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/type_inference_A17_t06 CompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/type_inference_A18_t01 CompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/type_inference_A18_t02 RuntimeError (expected Pass)
co19/LanguageFeatures/Patterns/type_inference_A18_t03 CompileTimeError (expected Pass)
co19/LanguageFeatures/Patterns/type_inference_A19_t02 CompileTimeError (expected Pass)
```
are failing on configurations
```
analyzer-asserts-linux
cfe-strong-linux
cfe-weak-linux
dart2js-hostasserts-strong-linux-x64-chrome
dart2js-hostasserts-weak-linux-x64-chrome
dart2wasm-hostasserts-linux-x64-d8
dartdevk-strong-linux-release-chrome
dartdevk-weak-linux-release-chrome
dartk-strong-linux-debug-x64
dartk-strong-linux-release-x64
dartk-weak-asserts-linux-debug-x64
dartk-weak-asserts-linux-release-x64
```

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

**Comment by sgrekhov** [comment]
Failure | Reason
-- | --
analyzer-asserts-linuxcfe...dart2js...dart2wasm-hostasserts-linux-x64-d8dartdevk...dartk... |  
new test -> CompileTimeError (expected Pass) |  
co19/LanguageFeatures/Patterns/type_inference_A11_t03 | https://github.com/dart-lang/language/issues/2845
co19/LanguageFeatures/Patterns/type_inference_A18_t01 | Fixed
cfe... |  
new test -> MissingCompileTimeError (expected Pass) |  
co19/LanguageFeatures/Patterns/shared_case_scope_A01_t04 | CFE issue. Patterns feature is not fully implemented yet
cfe...dart2js...dart2wasm-hostasserts-linux-x64-d8dartdevk...dartk... |  
new test -> CompileTimeError (expected Pass) |  
co19/LanguageFeatures/Patterns/type_inference_A11_t02 | Tools issue. Patterns feature is not fully implemented yet
co19/LanguageFeatures/Patterns/type_inference_A12_t01 |  Fixed
co19/LanguageFeatures/Patterns/type_inference_A13_t02 | Tools issue. Patterns feature is not fully implemented yet
co19/LanguageFeatures/Patterns/type_inference_A14_t03 | Tools issue. Patterns feature is not fully implemented yet
co19/LanguageFeatures/Patterns/type_inference_A16_t03 | https://github.com/dart-lang/language/issues/2845
co19/LanguageFeatures/Patterns/type_inference_A17_t05 | Tools issue. Patterns feature is not fully implemented yet
co19/LanguageFeatures/Patterns/type_inference_A18_t03 | Tools issue. Patterns feature is not fully implemented yet
co19/LanguageFeatures/Patterns/type_inference_A19_t02 | https://github.com/dart-lang/language/issues/2845
cfe-strong-linuxdart2js...dart2wasm-hostasserts-linux-x64-d8dartdevk...dartk... |  
new test -> CompileTimeError (expected Pass) |  
co19/LanguageFeatures/Patterns/type_inference_A13_t03 | Tools issue. Patterns feature is not fully implemented yet
co19/LanguageFeatures/Patterns/type_inference_A14_t02 | Tools issue. Patterns feature is not fully implemented yet
co19/LanguageFeatures/Patterns/type_inference_A16_t02 | Tools issue. Patterns feature is not fully implemented yet
co19/LanguageFeatures/Patterns/type_inference_A17_t06 | Tools issue. Patterns feature is not fully implemented yet
dart2js...dart2wasm-hostasserts-linux-x64-d8dartdevk...dartk... |  
new test -> RuntimeError (expected Pass) |  
co19/LanguageFeatures/Patterns/type_inference_A18_t02 | Tools issue. Patterns feature is not fully implemented yet

File: LanguageFeatures/Patterns/type_inference_A18_t01.dart [hunk 1]
```diff
@@ -30,17 +30,17 @@ main() {
...
-    v2.expectStaticType<Exactly<A>>();
-    v3.expectStaticType<Exactly<A>>();
+    v2.expectStaticType<Exactly<B>>();
+    v3.expectStaticType<Exactly<B>>();
   }
   {
-    final <String, A>{"key1": A v1, "key2": B v2, "key3": v3} =
-        <String, C>{"key1": d, "key2": d, "key3": d};
+    final <String, B>{"key1": A v1, "key2": B v2, "key3": v3} =
+        <String, D>{"key1": d, "key2": d, "key3": d};
```
Comment (eernstg) [inline LanguageFeatures/Patterns/type_inference_A18_t01.dart:41]: This one could remain unchanged as `<String, C>`, but `D` is fine, too!

**Review by eernstg — APPROVED** [review]
LGTM

**Comment by eernstg** [comment]
Thanks for triaging all the failures!

**Pull request merged**